pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -269,7 +288,30 @@ func SortNodesByUsage(nodes []NodeUsageMap) {
 	})
 }
 
-func CreateNodePodsMap(client clientset.Interface, nodes []*v1.Node) NodePodsMap {
+// sortPodsBasedOnPriority sorts pods based on priority and if their priorities are equal, they are sorted based on QoS tiers.
+func sortPodsBasedOnPriority(evictablePods []*v1.Pod) {
+	sort.Slice(evictablePods, func(i, j int) bool {
+		if evictablePods[i].Spec.Priority == nil && evictablePods[j].Spec.Priority != nil {
+			return true
+		}
+		if evictablePods[j].Spec.Priority == nil && evictablePods[i].Spec.Priority != nil {
+			return false
+		}
+		if (evictablePods[j].Spec.Priority == nil && evictablePods[i].Spec.Priority == nil) || (*evictablePods[i].Spec.Priority == *evictablePods[j].Spec.Priority) {
```

Comment:
@aveshagarwal - The above covers all the 3 scenarios when one of the pods doesn't have a priority and when both doesn't have priority. Please let me know, if you have anything else in your mind.

Reply:
this looks better for sure. The only thing I am not able to understand when can we hit (evictablePods[j].Spec.Priority == nil), can you explain the use cases?

Reply:
So, say we have 2 pods for comparing and first pod has priority where as 2nd one is not priority, we need this case.